pkg/provision/primitives/cache/cache.go

```diff
@@ -248,6 +249,90 @@ func (s *Fs) Exists(id string) (bool, error) {
 	return false, err
 }
 
+// NetworkExists exists checks if a network exists in cache already
+func (s *Fs) NetworkExists(name string) (bool, error) {
+	s.RLock()
+	defer s.RUnlock()
+
+	infos, err := ioutil.ReadDir(s.root)
+	if err != nil {
+		return false, err
+	}
+
+	for _, info := range infos {
+		if info.IsDir() {
+			continue
+		}
+
+		r, err := s.get(info.Name())
+		if err != nil {
+			return false, fmt.Errorf("failed get reservation: %w", err)
+		}
+
+		if r.Type == primitives.NetworkReservation {
+			nr := pkg.NetResource{}
+			if err := json.Unmarshal(r.Data, &nr); err != nil {
+				return false, fmt.Errorf("failed to unmarshal network from reservation: %w", err)
+			}
+
+			if nr.Name == name {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+// incrementCounters will increment counters for all workloads
+// for network workloads it will only increment those that have a unique name
+func (s *Fs) incrementCounters(statser provision.Statser) error {
+	s.RLock()
+	defer s.RUnlock()
+
+	uniqueNetworks := make(map[string]*provision.Reservation)
+
+	infos, err := ioutil.ReadDir(s.root)
+	if err != nil {
+		return err
+	}
+
+	for _, info := range infos {
+		if info.IsDir() {
+			continue
+		}
+
+		r, err := s.get(info.Name())
+		if err != nil {
+			return fmt.Errorf("failed get reservation: %w", err)
+		}
+
+		if r.Type == primitives.NetworkReservation {
+			nr := pkg.NetResource{}
+			if err := json.Unmarshal(r.Data, &nr); err != nil {
+				return fmt.Errorf("failed to unmarshal network from reservation: %w", err)
+			}
+
+			if _, ok := uniqueNetworks[nr.Name]; ok {
```

Comment:
not true, network name is not unique. the interface/bridges names are made unique by using a mashup of name and user id.

Reply:
Why did you resolve this? the Name is still not unique. you need to use the `name+id` key instead of just the name